src/Sarif.Multitool/ResultMatchSetCommand.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.Sarif.Baseline.ResultMatching;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool
+{
+    internal class ResultMatchSetCommand : CommandBase
+    {
+        private readonly IFileSystem _fileSystem;
+
+        public ResultMatchSetCommand(IFileSystem fileSystem = null)
+        {
+            _fileSystem = fileSystem ?? new FileSystem();
+        }
+
+        public int Run(ResultMatchSetOptions options)
+        {
+            options.OutputFolderPath = options.OutputFolderPath ?? Path.Combine(options.FolderPath, "Out");
+
+            ISarifLogMatcher matcher = ResultMatchingBaselinerFactory.GetDefaultResultMatchingBaseliner();
+            Formatting formatting = (options.PrettyPrint ? Formatting.Indented : Formatting.None);
+
+            // Remove previous results
+            if (Directory.Exists(options.OutputFolderPath))
+            {
+                Directory.Delete(options.OutputFolderPath, true);
+            }
+
+            // Create output folder
+            Directory.CreateDirectory(options.OutputFolderPath);
+
+            string previousFileName = "";
+            string previousGroup = "";
+            SarifLog previousLog = null, currentLog = null;
+
+            foreach (string filePath in Directory.GetFiles(options.FolderPath, "*.sarif"))
+            {
+                string fileName = Path.GetFileName(filePath);
+                string currentGroup = GetGroupName(fileName);
+
+                try
+                {
+                    currentLog = ReadSarifFile<SarifLog>(_fileSystem, filePath);
+
+                    // Compare each log with the previous one in the same group
+                    if (currentGroup.Equals(previousGroup) && currentLog?.Runs[0].Results.Count != 0 && previousLog?.Runs[0].Results.Count != 0)
+                    {
+                        Console.WriteLine();
+                        Console.WriteLine($"{previousFileName} -> {fileName}:");
+                        SarifLog result = matcher.Match(new[] { previousLog }, new[] { currentLog }).First();
```

Comment:
>result [](start = 33, length = 6)

Mostly I won't comment on the nits I find -- I'll just fix them and raise a PR -- but here I'll point out that "result" is an unfortunate variable name in a SARIF context, where the word means something specific. In my PR I propose `comparisonResultLog`; feel free to suggest other names.

Reply:
Oh, I see that when he passes `result` to `WriteDifferences`, the parameter name in `WriteDifferences` is "`mergedLog`". I like that. I'll use it.

---
In reply to: [296050568](https://github.com/microsoft/sarif-sdk/pull/1539#discussion_r296050568) [](ancestors = 296050568)